### PR TITLE
add cmake script to choose benchmark apps

### PR DESCRIPTION
**Pull request by malus-brandywine**
While developing/debugging application, its very useful to run only
the application that a developer works on. It reduces the amount of
output and helps seeing debug custom messages.

Signed-off-by: Nataliya Korovkina <malus.brandywine@gmail.com>

### CODE REVIEW

**Comment by axel-h** [comment]
Having this explicitly seems a developer friendly thing to me

File: benchmark_list.cmake [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# Copyright 2022, Nataliya Korovkina <malus.brandywine@gmail.com>
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+# Set the list of benchmark applications to be included into the image
+# Uncomment command set() when needed
+
+# Application to benchmark seL4 IPC
+# default AppIpcBench is ON
```
Comment (axel-h) [inline benchmark_list.cmake:11]: could be a bit less verbose/redundant everywhere
```suggestion
# default is ON
```

Comment (malus-brandywine) [reply benchmark_list.cmake:11]: Agreed, I'll change it

**Comment by kent-mcleod** [comment]
Instead of adding a new file for this, I'd suggest adding it to easy-settings.cmake which is already supposed to expose project specific config options like which application to benchmark.  The added benefit is there is the griddle script that knows how to turn what's in easy-settings.cmake into a command line help message + build initialization:
```
~/workspaces/sel4bench/build$ ../griddle --help
griddle: easy cooking with CMake

griddle eases the setup of seL4-related builds by exposing only the most
commonly-used configuration variables in the seL4 CMake infrastructure.  These
differ between projects, but you can always discover them with:
    griddle --help

Usage:
    griddle [--compiler={gcc|llvm}] [CMAKE-CONFIGURATION-VARIABLE] ...
    griddle --help
    griddle --platform-help

Options:
    --compiler={gcc|llvm}   Report "gcc" or "llvm" compiler suite to CMake.
                            (default: gcc)
    --help                  Display this message and exit.
    --platform-help         List supported target platforms and exit.

               Option    Type   Default  Description

            --RELEASE    BOOL        ON  Performance optimized build
           --HARDWARE    BOOL       OFF  Configuration for sel4bench hardware app
              --FAULT    BOOL       OFF  Configuration sel4bench fault app
               --VCPU    BOOL       OFF  Whether or not to run the VCPU benchmarks
                --SMP    BOOL       OFF  Configuration sel4bench smp app
           --PLATFORM  STRING  "x86_64"  Platform to test
           --FASTPATH    BOOL        ON  Turn fastpath on or off
            --ARM_HYP    BOOL       OFF  ARM EL2 hypervisor features on or off
                --MCS    BOOL       OFF  MCS kernel
        --AppIpcBench    BOOL        ON  Application to benchmark seL4 IPC
        --AppIrqBench    BOOL       OFF  Application to benchmark seL4 IRQs from inside the kernel
    --AppIrqUserBench    BOOL        ON  Application to benchmark seL4 IRQs without modification to the kernel
  --AppSchedulerBench    BOOL        ON  Application to benchmark seL4 scheduler without modification to the kernel
     --AppSignalBench    BOOL        ON  Application to benchmark seL4 signals without modification to the kernel
        --AppSmpBench    BOOL       OFF  Enable SMP benchmarks
       --AppVcpuBench    BOOL       OFF  Application to benchmark seL4 VCPU performance
--AppPageMappingBench    BOOL        ON  Application to benchmark seL4 mapping a series of pages
       --AppSyncBench    BOOL        ON  Application to benchmark seL4 sync
      --AppFaultBench    BOOL       OFF  Application to benchmark seL4 faults without modification to the kernel
   --AppHardwareBench    BOOL       OFF  Application to benchmark hardware-related operations
```
From this diff:
```

```
diff --git a/easy-settings.cmake b/easy-settings.cmake
index bf85210..34a1bf6 100644
--- a/easy-settings.cmake
+++ b/easy-settings.cmake
@@ -28,3 +28,16 @@ set(PLATFORM "x86_64" CACHE STRING "Platform to test")
 set(FASTPATH ON CACHE BOOL "Turn fastpath on or off")
 set(ARM_HYP OFF CACHE BOOL "ARM EL2 hypervisor features on or off")
 set(MCS OFF CACHE BOOL "MCS kernel")
+
+set(AppIpcBench ON CACHE BOOL "Application to benchmark seL4 IPC")
+# see apps/irq/CMakeLists.txt for current state of the benchmark
+set(AppIrqBench OFF CACHE BOOL "Application to benchmark seL4 IRQs from inside the kernel")
+set(AppIrqUserBench ON CACHE BOOL "Application to benchmark seL4 IRQs without modification to the kernel")
+set(AppSchedulerBench ON CACHE BOOL "Application to benchmark seL4 scheduler without modification to the kernel")
+set(AppSignalBench ON CACHE BOOL "Application to benchmark seL4 signals without modification to the kernel")
+set(AppSmpBench OFF CACHE BOOL "Enable SMP benchmarks")
+set(AppVcpuBench OFF CACHE BOOL "Application to benchmark seL4 VCPU performance")
+set(AppPageMappingBench ON CACHE BOOL "Application to benchmark seL4 mapping a series of pages")
+set(AppSyncBench ON CACHE BOOL "Application to benchmark seL4 sync")
+set(AppFaultBench OFF CACHE BOOL "Application to benchmark seL4 faults without modification to the kernel")
+set(AppHardwareBench OFF CACHE BOOL "Application to benchmark hardware-related operations")
```

**Comment by malus-brandywine** [comment]
@kent-mcleod , thank you for the hint!

**Comment by malus-brandywine** [comment]
Is that technically possible to skip the 1st & 2nd commits and merge the two recent ones?

**Comment by lsf37** [comment]
> Is that technically possible to skip the 1st & 2nd commits and merge the two recent ones?

It's easy to squash everything into one commit when we merge the pull request to get the overall diff.

**Comment by axel-h** [comment]
> Is that technically possible to skip the 1st & 2nd commits and merge the two recent ones?

Technically, the PR is tied to the branch name only, so you can reorder commits and/or drop specific ones and just force-push.

**Comment by malus-brandywine** [comment]
Thanks, @axel-h !

**Review by kent-mcleod — APPROVED** [review]

**Comment by kent-mcleod** [comment]
Thanks, @malus-brandywine !

**Pull request merged**